src/core/bpf/allow_bind/allow-bind.c

```diff
@@ -0,0 +1,42 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "maps.h"
+
+#include <bpf/bpf_endian.h>
+
+#include <linux/bpf.h>
+#include <netinet/in.h>
+
+static inline int allow_bind_impl(
+                struct bpf_sock_addr* ctx,
+                struct bpf_map_def* allowed_ports) {
+        __u16 port_key;
+        __u8* port;
+
```

Comment:
assert(ctx)

Reply:
and allowed_ports, unless it's fine if it's null

Reply:
oh wait this is the bpf part, can we assert there?

Reply:
@bluca 

No, we can't. llvm compiles the code into elf file, then it's loaded into kernel [[0]](https://docs.cilium.io/en/v1.9/bpf/#llvm)
It's up to BPF verifier to make to ensure that the code is safe and won't crash the kernel.
[0] https://docs.cilium.io/en/v1.9/bpf/#llvm